akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1707,9 +1707,12 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
    * streams, one for each element key. The key is computed for each element
    * using the given function. When a new key is encountered for the first time
    * a new substream is opened and subsequently fed with all elements belonging to
-   * that key. If `allowClosedSubstreamRecreation` is set to `true` a substream
-   * belonging to a specific key will be recreated if it was closed before,
-   * otherwise elements belonging to that key will be dropped.
+   * that key.
+   *
+   * WARNING: If `allowClosedSubstreamRecreation` is set to `false` (default behavior) the stage
+   * keeps track of all keys of streams that have already been closed. If you expect an infinite
+   * number of keys this can cause memory issues. Elements belonging to those keys are drained
+   * directly and not send to the substream.
    *
```

Comment:
please also add the warning to the normal, without the boolean param, method; since there this WARNING behavior will always be the one

Reply:
Ah sure, that makes lots of sense! Changed and pushed!